src/commands/commit/upload.ts

```diff
@@ -0,0 +1,182 @@
+import retry from 'async-retry'
+import chalk from 'chalk'
+import {Command} from 'clipanion'
+import {BufferedMetricsLogger} from 'datadog-metrics'
+
+import {apiConstructor} from './api'
+import {ApiKeyValidator} from './apikey'
+import {InvalidConfigurationError} from './errors'
+import {getRepositoryData, newSimpleGit, RepositoryData} from './git'
+import {APIHelper, Payload} from './interfaces'
+import {getMetricsLogger} from './metrics'
+import {
+  renderCommandInfo,
+  renderConfigurationError,
+  renderFailedUpload,
+  renderRetriedUpload,
+  renderSuccessfulCommand,
+} from './renderer'
+import {getBaseIntakeUrl} from './utils'
+
+const errorCodesNoRetry = [400, 403, 413]
+
+export class UploadCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Report the current commit details to Datadog.',
+    details: `
+            This command will upload the commit details to Datadog in order to create links to your repositories inside DataDog's UI.
+            See README for details.
+        `,
+    examples: [['Upload the current commit details', 'datadog-ci report-commits upload']],
+  })
+
+  public repositoryURL?: string
+
+  private apiKeyValidator: ApiKeyValidator
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY,
+    datadogSite: process.env.DATADOG_SITE || 'datadoghq.com',
+  }
+  private dryRun = false
+
+  constructor() {
+    super()
+    this.apiKeyValidator = new ApiKeyValidator(this.config.apiKey, this.config.datadogSite)
+  }
+
+  public async execute() {
+    const initialTime = new Date().getTime()
+    const api = this.getApiHelper()
+    this.context.stdout.write(renderCommandInfo(this.dryRun))
+
+    const cliVersion = require('../../../package.json').version
+    const metricsLogger = getMetricsLogger(cliVersion)
+    const payload = await this.getPayloadToUpload(cliVersion)
+    if (payload === undefined) {
```

Comment:
You can replace all `abc === undefined` with `!abc` and `return undefined` with `return`
```suggestion
    if (!payload) {
```